processor/transformprocessor/internal/traces/traces.go

```diff
@@ -215,7 +215,7 @@ func accessInstrumentationScopeVersion() pathGetSetter {
 func accessTraceID() pathGetSetter {
 	return pathGetSetter{
 		getter: func(ctx common.TransformContext) interface{} {
-			return ctx.GetItem().(ptrace.Span).TraceID()
+			return ctx.GetItem().(ptrace.Span).TraceID().HexString()
```

Comment:
Instead of this (it allocates extra string) can we determine that it checks for "trace_id" and convert that to TraceID? Or maybe change the query to be `span_id == span_id("0102030405060708")`?

Reply:
@bogdandrutu opened https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/10487 to add SpanIDs and TraceIDs to the grammar.  Once that PR is merged, we'd be able to have the accessor's Getter return the literal SpanID/TraceID and Setter accept a literal SpanID/TraceID instead of a string.

Reply:
Should this be updated now that #10487 merged?

Reply:
Yes I haven't gotten back to the PR yet, got pulled away but some other work.  I plan to work on it today.

Reply:
Done.  Traces now returns/accepts SpanID/TraceID in the accessors.